Regression in Eigen detection on Ubuntu 14.04

With a fresh checkout from `develop`:
```
-- Looking for Eigen3...
-- Looking for Eigen3... - not found
CMake Error at CMake/Basis/CommonTools.cmake:801 (message):
  

  Module Numerics requires Eigen3.  Please ensure that the package is
  installed in a standard system location or set DEPENDS_Eigen3_DIR to the
  installation prefix (i.e., root directory of the installation).

  The DEPENDS_Eigen3_DIR variable can alternatively be set to the directory
  containing a Eigen3Config.cmake or eigen3-config.cmake file.  If no such
  file exists, contact either the developer of this project or CMake BASIS to
  provide a FindEigen3.cmake file.

Call Stack (most recent call first):
  CMake/Basis/ProjectTools.cmake:2008 (basis_find_package)
  CMake/Basis/ProjectTools.cmake:2288 (basis_find_packages)
  CMake/Modules/mirtkProjectBegin.cmake:90 (basis_project_begin)
  CMake/Modules/mirtkConfigureModule.cmake:41 (mirtk_project_begin)
  Modules/Numerics/CMakeLists.txt:20 (mirtk_configure_module)


-- Configuring incomplete, errors occurred!
See also "/homes/gvaillan/workspace/MIRTK/build/CMakeFiles/CMakeOutput.log".
```

Whereas with `master`:
```
-- Looking for Eigen3...
-- Found Eigen3: /homes/gvaillan/workspace/MIRTK/ThirdParty/Eigen (Required is at least version "2.91.0") 
-- Looking for Eigen3... - found
```

Still not working:
```
-- Looking for Eigen3...
-- Looking for Eigen3... - not found
CMake Error at CMake/Basis/CommonTools.cmake:824 (message):
  

  Module Numerics requires Eigen3.  Please ensure that the package is
  installed in a standard system location or set DEPENDS_Eigen3_DIR to the
  installation prefix (i.e., root directory of the installation).

  The DEPENDS_Eigen3_DIR variable can alternatively be set to the directory
  containing a Eigen3Config.cmake or eigen3-config.cmake file.  If no such
  file exists, contact either the developer of this project or CMake BASIS to
  provide a FindEigen3.cmake file.

Call Stack (most recent call first):
  CMake/Basis/ProjectTools.cmake:2008 (basis_find_package)
  CMake/Basis/ProjectTools.cmake:2288 (basis_find_packages)
  CMake/Modules/mirtkProjectBegin.cmake:90 (basis_project_begin)
  CMake/Modules/mirtkConfigureModule.cmake:41 (mirtk_project_begin)
  Modules/Numerics/CMakeLists.txt:20 (mirtk_configure_module)


-- Configuring incomplete, errors occurred!
```
with the most recent tip of `develop` (a01f138ab01218c84d8becd83b881d2a2eed1d78)

--- a/CMake/Basis/CommonTools.cmake
+++ b/CMake/Basis/CommonTools.cmake
@@ -54,7 +54,7 @@ macro (find_package)
   list (APPEND _BASIS_FIND_LIBRARY_SUFFIXES "{${CMAKE_FIND_LIBRARY_SUFFIXES}}")
   list (APPEND _BASIS_FIND_EXECUTABLE_SUFFIX "${CMAKE_FIND_EXECUTABLE_SUFFIX}")
   _find_package(${ARGV})
-  # map obsolete <PKG>_* variables to case-sensitive <Pkg>_*
+  # map common uppercase <PKG>_* variables to case-preserving <Pkg>_*
   string (TOUPPER "${ARGV0}" _FP_ARGV0_U)
   foreach (_FP_VAR IN ITEMS FOUND DIR USE_FILE
                             VERSION VERSION_STRING
@@ -63,7 +63,7 @@ macro (find_package)
                             INCLUDE_DIR INCLUDE_DIRS INCLUDE_PATH
                             LIBRARY_DIR LIBRARY_DIRS LIBRARY_PATH
                             EXECUTABLE COMPILER CONVERTER)
-    if (NOT DEFINED ${ARGV0}_${_FP_VAR} AND DEFINED ${_FP_ARGV0_U}_${_FP_VAR})
+    if (NOT ${ARGV0}_${_FP_VAR} AND DEFINED ${_FP_ARGV0_U}_${_FP_VAR})
       set (${ARGV0}_${_FP_VAR} "${${_FP_ARGV0_U}_${_FP_VAR}}")
     endif ()
   endforeach ()

